aws/terraform/README.md

```diff
@@ -0,0 +1,112 @@
+## Pre-requisites
+
+### Terraform
+
+[Download](https://www.terraform.io/downloads.html) and install terraform 0.14.x.
+
+### AWS
+
+AWS should be [configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html) with a profile. The `AWS_PROFILE` environment should be set to the profile that has been configured.
+
+The [awscli](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html) can be used as follows to 
+confirm that it has been configured properly by running:
+
+```
+aws sts get-caller-identity
+```
+
+which should output your account information.
+
+## Setup
+
+### Infrastructure stack
+
+The infra sub-project provides some pre-requisite infrastructure for the metaflow service. For more details see the [README](aws/terraform/infra/README.md)
+
+Copy `example.tfvars` to `prod.tfvars` (or whatever environment name you prefer) and update that `env` name and the `region` as needed.
+
+Initialize the terraform:
+
+`cd infra && terraform init`
+
+Apply it:
+
+```
+terraform apply --var-file prod.tfvars
+```
+
+### Metaflow stack
+
+The metaflow sub-project provisions the metadata API, step functions, and an AWS Batch queue. For more details see the 
+[README](aws/terraform/metaflow/README.md)
+
+Copy `example.tfvars` to `prod.tfvars` (or whatever environment name you prefer) and update that `env` name and the `region` as needed.
+Metadata API authentication: the endpoint is exposed to the public internet via API Gateway, but only accessible to the IPs that match `access_list_cidr_blocks` (default is none).
+Additionally:
+* There are variables which govern the three compute environments associated with the AWS Batch queue that can be adjusted based on needs. 
+* The `enable_step_functions` flag can be set to false to not provision the step functions infrastructure. 
+* The `access_list_cidr_blocks` should be set to the network cidr blocks that will be accessing the metadata API.
+
+Initialize the terraform:
+
+`cd infra && terraform init`
+
+Apply it:
+
+```
+terraform apply --var-file prod.tfvars
+```
+
+Once the terraform executes, grab the output of `metaflow_profile_json` and put it in `~/.metaflowconfig/config_prod.json`
+
+Next set the `METAFLOW_PROFILE` environment variable to `prod` to tell the metaflow client which configuration to use.
+
+### Default Batch Image
+
+The generated Metaflow batch configuration sets `METAFLOW_BATCH_CONTAINER_IMAGE` and `METAFLOW_BATCH_CONTAINER_REGISTRY` to an ECR repository. The Metaflow batch image must be pushed into the repository before the first flow can be executed.
```

Comment:
Can we make this optional so that the user can choose if they want the ECR repository to be set up or not?

Reply:
I don't think this would be a problem, @kldavis4 do you forsee any issue that I'm not grasping?

Reply:
yeah, I think the dockerhub thing was the issue we were running into. I'm totally fine with making it optional